classes/02class/exercises/c02-network02/ramiray/c02-network02.txt

```diff
@@ -0,0 +1,16 @@
+1.10.0.0.0/16
+
+    Subnet        Network 		    Host Range         Usage Hosts    Broadcasts       AWS Reserved(1st 4 ips and last 1 ip)
+subnet a    10.0.0.0/21		10.0.0.1 - 10.0.7.254	    2046		10.0.7.255		10.0.0.1,10.0.0.2,10.0.0.3,10.0.0.4,10.0.7.254			
+subnet b    10.0.8.0/21		10.0.8.1 - 10.0.15.254	    2046		10.0.15.255     10.0.8.1,10.0.8.2,10.0.8.3,10.0.8.4,10.0.15.254
+subnet c    10.0.16.0/20	10.0.16.1 - 10.0.31.254	    4094		10.0.31.255     10.0.16.1,10.0.16.2,10.0.16.3,10.0.16.4,10.0.31.254
+subnet d    10.0.32.0/19	10.0.32.1 - 10.0.63.254	    8190		10.0.63.255     10.0.32.1,10.0.32.2,10.0.32.3,10.0.32.4,10.0.63.254
+
+
+2. 192.168.0.0/24
+
+    Subnet          Network 		    Host Range           Usage Hosts    Broadcasts       AWS Reserved(1st 4 ips and last 1 ip)
```

Comment:
Every network has 2 reserved IPs: the first one (called "network id") and the last one (called "broadcast") -- that is nothing to do with AWS.

For example with `192.168.0.0/26`:
- Network ID: `192.168.0.0`
- Broadcast: `192.168.0.63`
- Total of usable hosts: 62

In AWS they also reserve 3 more IPs for "gateway", "dns" and "future use".
From the example above:
- Gateway: `192.168.0.1`
- DNS: `192.168.0.2`
- Reserved for future use: `192.168.0.3`
- Total hosts usable in AWS: 59

So in the last column you can list only the 3 IPs reserved for AWS. Note the `192.168.0.4` for example is usable in AWS (if you consider the first 4, the first IP will be the network ID `192.168.0.0`)

Feel free to ask me more if that is not clear